### PR TITLE
Ensure that exact synonyms in NCIT are preserved

### LINKED ISSUE #612: Synonyms missing from `components/ncit.owl`
## Overview
While working on the synonym sync, [I noticed](https://github.com/monarch-initiative/mondo-ingest/pull/93#discussion_r1667202320) that this file has no synonyms.

## Additional info
`tmp/component-download-ncit.owl.owl` does have synonyms, though.

<details><summary>Example case code snippets</summary>
<p>

Example: `NCIT:C157344`

`components/ncit.owl`:
```owl
    <owl:Class rdf:about="http://purl.obolibrary.org/obo/NCIT_C101540">
        <rdfs:subClassOf rdf:resource="http://purl.obolibrary.org/obo/NCIT_C87543"/>
        <oboInOwl:inSubset rdf:resource="http://purl.obolibrary.org/obo/NCIT_C116977"/>
        <oboInOwl:inSubset rdf:resource="http://purl.obolibrary.org/obo/NCIT_C118168"/>
        <rdfs:label>Stage I Differentiated Thyroid Gland Carcinoma 45 Years and Older AJCC v7</rdfs:label>
    </owl:Class>
```

`tmp/component-download-ncit.owl.owl`:
```owl
    <owl:Class rdf:about="http://purl.obolibrary.org/obo/NCIT_C157344">
        <rdfs:subClassOf rdf:resource="http://purl.obolibrary.org/obo/NCIT_C28681"/>
        <obo:IAO_0000115>A preparation of autologous T-lymphocytes transduced with a retroviral vector encoding a T-cell receptor (TCR) sequence specific for CT-RCC-1, a tumor-associated antigen (TAA) and HLA-A11-restricted peptide encoded by human endogenous retrovirus (HERV) type E as well as a truncated CD34 chain (CD34t), with potential antineoplastic activity. Upon isolation, transduction, expansion ex vivo and re-introduction into the patient, the autologous CT-RCC-1 HERV-E-TCR-transduced-HLA-A11-restricted CD8+/CD34t+ T-cells bind to and induce selective toxicity in tumor cells expressing both the HLA-A11 allele and the CT-RCC-1 HERV-E antigen. The CD34t protein allows the transduced cells to be identified with an anti-CD34 antibody, and facilitates monitoring of the genetically modified T-cells following adoptive transfer. CT-RCC-1 HERV-E is a TAA found in a high percentage of clear cell renal cell carcinoma (ccRCC) cells.</obo:IAO_0000115>
        <obo:NCIT_A7 rdf:resource="http://purl.obolibrary.org/obo/NCIT_C178664"/>
        <obo:NCIT_NHC0>C157344</obo:NCIT_NHC0>
        <obo:NCIT_P106>Cell</obo:NCIT_P106>
        <obo:NCIT_P106>Pharmacologic Substance</obo:NCIT_P106>
        <obo:NCIT_P107>Autologous CT-RCC-1 HERV-E-TCR-transduced-HLA-A11-restricted CD8+/CD34t+ T-cells</obo:NCIT_P107>
        <obo:NCIT_P108>Autologous CT-RCC-1 HERV-E-TCR-transduced-HLA-A11-restricted CD8+/CD34t+ T-cells</obo:NCIT_P108>
        <obo:NCIT_P208>CL936969</obo:NCIT_P208>
        <obo:NCIT_P322>CTRP</obo:NCIT_P322>
        <obo:NCIT_P329>796813</obo:NCIT_P329>
        <obo:NCIT_P330>796813</obo:NCIT_P330>
        <obo:NCIT_P399>796813</obo:NCIT_P399>
        <oboInOwl:hasExactSynonym>Anti-CT-RCC-1 HERV-E-TCR-transduced-HLA-A11-restricted Autologous CD8+/CD34t+ T-cells (SY)</oboInOwl:hasExactSynonym>
        <oboInOwl:hasExactSynonym>Autologous CT-RCC-1 HERV-E-TCR-transduced-HLA-A11-restricted CD8+/CD34t+ T-cells</oboInOwl:hasExactSynonym>
        <oboInOwl:hasExactSynonym>Autologous CT-RCC-1 HERV-E-TCR-transduced-HLA-A11-restricted CD8+/CD34t+ T-lymphocytes</oboInOwl:hasExactSynonym>
        <oboInOwl:inSubset rdf:resource="http://purl.obolibrary.org/obo/NCIT_C116977"/>
        <oboInOwl:inSubset rdf:resource="http://purl.obolibrary.org/obo/NCIT_C116978"/>
        <oboInOwl:inSubset rdf:resource="http://purl.obolibrary.org/obo/NCIT_C128784"/>
        <oboInOwl:inSubset rdf:resource="http://purl.obolibrary.org/obo/NCIT_C176424"/>
        <rdfs:label>Autologous CT-RCC-1 HERV-E-TCR-transduced-HLA-A11-restricted CD8+/CD34t+ T-cells</rdfs:label>
    </owl:Class>
    <owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/NCIT_C157344"/>
        <owl:annotatedProperty rdf:resource="http://purl.obolibrary.org/obo/IAO_0000115"/>
        <owl:annotatedTarget>A preparation of autologous T-lymphocytes transduced with a retroviral vector encoding a T-cell receptor (TCR) sequence specific for CT-RCC-1, a tumor-associated antigen (TAA) and HLA-A11-restricted peptide encoded by human endogenous retrovirus (HERV) type E as well as a truncated CD34 chain (CD34t), with potential antineoplastic activity. Upon isolation, transduction, expansion ex vivo and re-introduction into the patient, the autologous CT-RCC-1 HERV-E-TCR-transduced-HLA-A11-restricted CD8+/CD34t+ T-cells bind to and induce selective toxicity in tumor cells expressing both the HLA-A11 allele and the CT-RCC-1 HERV-E antigen. The CD34t protein allows the transduced cells to be identified with an anti-CD34 antibody, and facilitates monitoring of the genetically modified T-cells following adoptive transfer. CT-RCC-1 HERV-E is a TAA found in a high percentage of clear cell renal cell carcinoma (ccRCC) cells.</owl:annotatedTarget>
        <obo:NCIT_P378>NCI</obo:NCIT_P378>
    </owl:Axiom>
    <owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/NCIT_C157344"/>
        <owl:annotatedProperty rdf:resource="http://www.geneontology.org/formats/oboInOwl#hasExactSynonym"/>
        <owl:annotatedTarget>Anti-CT-RCC-1 HERV-E-TCR-transduced-HLA-A11-restricted Autologous CD8+/CD34t+ T-cells (SY)</owl:annotatedTarget>
        <obo:NCIT_P383>SY</obo:NCIT_P383>
        <obo:NCIT_P384>NCI</obo:NCIT_P384>
    </owl:Axiom>
    <owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/NCIT_C157344"/>
        <owl:annotatedProperty rdf:resource="http://www.geneontology.org/formats/oboInOwl#hasExactSynonym"/>
        <owl:annotatedTarget>Autologous CT-RCC-1 HERV-E-TCR-transduced-HLA-A11-restricted CD8+/CD34t+ T-cells</owl:annotatedTarget>
        <obo:NCIT_P383>DN</obo:NCIT_P383>
        <obo:NCIT_P384>CTRP</obo:NCIT_P384>
    </owl:Axiom>
    <owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/NCIT_C157344"/>
        <owl:annotatedProperty rdf:resource="http://www.geneontology.org/formats/oboInOwl#hasExactSynonym"/>
        <owl:annotatedTarget>Autologous CT-RCC-1 HERV-E-TCR-transduced-HLA-A11-restricted CD8+/CD34t+ T-cells</owl:annotatedTarget>
        <obo:NCIT_P383>PT</obo:NCIT_P383>
        <obo:NCIT_P384>NCI</obo:NCIT_P384>
    </owl:Axiom>
    <owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/NCIT_C157344"/>
        <owl:annotatedProperty rdf:resource="http://www.geneontology.org/formats/oboInOwl#hasExactSynonym"/>
        <owl:annotatedTarget>Autologous CT-RCC-1 HERV-E-TCR-transduced-HLA-A11-restricted CD8+/CD34t+ T-lymphocytes</owl:annotatedTarget>
        <obo:NCIT_P383>SY</obo:NCIT_P383>
        <obo:NCIT_P384>NCI</obo:NCIT_P384>
    </owl:Axiom>
```

Here's the component goal. At a glance, I don't see why synonyms are being stripped, but I haven't looked closely:
```make
$(COMPONENTSDIR)/ncit.owl: $(TMPDIR)/ncit_relevant_signature.txt | component-download-ncit.owl
	if [ $(SKIP_HUGE) = false ] && [ $(COMP) = true ]; then $(ROBOT) remove -i $(TMPDIR)/component-download-ncit.owl.owl --select imports \
		rename --mappings config/property-map.sssom.tsv --allow-missing-entities true --allow-duplicates true \
		query --update ../sparql/rm_xref_by_prefix.ru \
		remove -T $(TMPDIR)/ncit_relevant_signature.txt --select complement --select "classes individuals" --trim false \
		remove -T config/properties.txt --select complement --select properties --trim true \
		remove --term "http://purl.obolibrary.org/obo/NCIT_C179199" --axioms "equivalent" \
		annotate --ontology-iri $(URIBASE)/mondo/sources/ncit.owl --version-iri $(URIBASE)/mondo/sources/$(TODAY)/ncit.owl -o $@; fi
```

</p>
</details>

### PR DESCRIPTION
**Pull request by matentzn**
Resolves #612

## Overview
This works because it prevents that `remove -T config/properties.txt --select complement --select properties --trim true \` removes synonyms (they are removed because of these axiom annotations).

### Documentation
Was the documentation added/updated under `docs/`?
- [ ] Yes
- [ ] No, updates to the docs were not necessary after careful consideration

### QC
Was the full pipeline run before submitting this PR using `sh run.sh make build-mondo-ingest` on this branch (after 
`docker pull obolibrary/odkfull:dev`), and no errors occurred?
- [ ] Yes
- [ ] No, there are no functional (code-related) changes to the pipeline in the PR, so no re-run is necessary
   
### New Packages
Were any new *Python packages* added?
- [ ] Yes, [requirements.txt.full](https://github.com/INCATools/ontology-development-kit/blob/master/requirements.txt.full) was updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [x] No

Were any other *non-Python packages* added?
- [ ] Yes, [Dockerfile](https://github.com/INCATools/ontology-development-kit/blob/master/Dockerfile) updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [x] No

### PR Review and Conversations Resolved
Has the PR been sufficiently reviewed by at least 1 team member of the Mondo Technical team and all threads resolved?
- [ ] Yes

### CODE REVIEW

File: src/ontology/mondo-ingest.Makefile [hunk 1]
```diff
@@ -96,6 +96,9 @@ $(COMPONENTSDIR)/ncit.owl: $(TMPDIR)/ncit_relevant_signature.txt | component-dow
 		rename --mappings config/property-map.sssom.tsv --allow-missing-entities true --allow-duplicates true \
 		query --update ../sparql/rm_xref_by_prefix.ru \
 		remove -T $(TMPDIR)/ncit_relevant_signature.txt --select complement --select "classes individuals" --trim false \
+			--drop-axiom-annotations NCIT:P378 \
+			--drop-axiom-annotations NCIT:P383 \
+			--drop-axiom-annotations NCIT:P384 \
```
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:101]: I see the synonyms and I do these axioms on them. But I haven't fully wrapped my mind (yet) around how this robot remove command would be removing those synonyms or how this solution works.

Comment (matentzn) [reply src/ontology/mondo-ingest.Makefile:101]: Sorry I didnt have time last night to neither make a proper PR nor explain it.

In the next step

```
		remove -T config/properties.txt --select complement --select properties --trim true \
```

removes all properties from NCIT which are not in `config/properties.txt`. Or I should say it more brutally:

It removes _all axioms in which a property exist that is not in this list_.

NCIT:P383, for example, is not in `config/properties.txt`.

Now here is the key (and the problem)

If you find this axiom:

```
id: MONDO:123
synonym: "a" [NCIT:P383="YO"]
```

ROBOT recognises NCIT:P383 part of the axiom.. and deletes it.

Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile:101]: Ahhh... I see. In RDF/XML, which I'm most familiar, it looks like the synonym is "just an annotation property", and that the axiom for NCIT:P383 is "a separate axiom on that property".

But I see that way of modeling it / thinking about it is not entirely correct, and that's not how robots sees it either.

Comment (twhetzel) [reply src/ontology/mondo-ingest.Makefile:101]: Interesting. I saw that nci had the synonyms represented in two ways in the class:
1) oio:hasExactSynonym as:`<oboInOwl:hasExactSynonym>Refractory Mucinous Carcinoma Peritonei</oboInOwl:hasExactSynonym>`
2) separate axioms on the class
```
<owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/NCIT_C179180"/>
        <owl:annotatedProperty rdf:resource="http://www.geneontology.org/formats/oboInOwl#hasExactSynonym"/>
        <owl:annotatedTarget>Refractory Mucinous Carcinoma Peritonei</owl:annotatedTarget>
        <obo:NCIT_P383>SY</obo:NCIT_P383>
        <obo:NCIT_P384>NCI</obo:NCIT_P384>
    </owl:Axiom>
```

Are you saying that ROBOT removes `NCIT_P383` and also then removes `hasExactSynonym` as a downstream effect?

Comment (matentzn) [reply src/ontology/mondo-ingest.Makefile:101]: Exactly, and the reason for this is that the owl api will parse this RDFXML structure into a single owlaxiom instance with two axiom annotations!

Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile:101]: What trish copy/pasted is exactly what I observed as well.

Hmm, interesting. @matentzn Forgive my tangent, but as our feedback is also probably valued for ROBOT, and OWL API (which I guess we have a whole lot less sway over), do you think we are unique in being surprised by this behavior? Or do you think that what OWL API is doing here is poor design? Or are computational ontologies so complex that confusing cases like this are just inevitable?

Comment (twhetzel) [reply src/ontology/mondo-ingest.Makefile:101]: Joe - can you ask this in Slack? It's too much of a tangent for this PR. Thank you.

**Comment by twhetzel** [comment]
I am running the test build on this now.

**Comment by twhetzel** [comment]
Test data build for this branch was reviewed and approved https://github.com/monarch-initiative/mondo-ingest/pull/631

**Pull request merged**